test/mailers/user_mailer_test.rb

```diff
@@ -24,4 +24,17 @@ class UserMailerTest < ActionMailer::TestCase
     assert_equal email.to, [@user.email]
     assert_in_delta @subdomain.analytics_report_last_sent, Time.zone.now.at_beginning_of_day, 1
   end
+
+  test 'sends OTP in email to user when 2fa is enabled' do
+    Subdomain.current.update(enable_2fa: true)
+
+    assert_emails 1 do
+      UserMailer.send_otp(@user).deliver_later
+    end
+
+    last_email  = ActionMailer::Base.deliveries.last
+
+    assert_equal "OTP", last_email.subject
+    assert_equal last_email.to, [@user.email]
+  end
```

Comment:
@ajaystha42 please add an assertion to verify the subdomain name + link and the OTP code in the email body